From 7299db08add19dc0f944e1099240b28c1602e011 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Wed, 9 Sep 2020 22:24:38 +0530
Subject: [PATCH 0764/1921] octeontx2-pf: Return proper RSS indirection table
 size always

With current logic, if the interface is not initialized even once
then size of indirection table returned is zero. And ethtool says
"Operation not supported". Fixed this by returning the size of
indirection table array.

Change-Id: I1b9a394f66c14d65492af20bbc233da44237e087
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/35638
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index 68156fd0250e..2b4aab207c22 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -933,7 +933,7 @@ static u32 otx2_get_rxfh_indir_size(struct net_device *dev)
 {
 	struct otx2_nic *pfvf = netdev_priv(dev);
 
-	return pfvf->hw.rss_info.rss_size;
+	return sizeof(pfvf->hw.rss_info.ind_tbl);
 }
 
 /* Get RSS configuration*/
-- 
2.31.1

